build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -54,27 +52,33 @@ impl Goodie for GraalVM {
 
     fn is_active(&self) -> BoxFuture<'static, Result<bool>> {
         let expected_graal_version = self.graal_version.clone();
-        let expected_java_language_version = self.java_version;
         async move {
             let found_version = find_graal_version().await?;
             ensure!(found_version == expected_graal_version, "GraalVM version mismatch. Expected {expected_graal_version}, found {found_version}.");
-
-            let found_java_version = Java.check_language_version().await?;
-            ensure!(
-                found_java_version == expected_java_language_version,
-                "Java language version mismatch. Expected {expected_java_language_version}, found {found_java_version}."
-            );
-
             Result::Ok(true)
         }
         .boxed()
     }
 
     fn activation_env_changes(&self, package_path: &Path) -> Result<Vec<crate::env::Modification>> {
-        let package_path = package_path.join(self.root_directory_name());
+        let dir_entries = package_path
+            .read_dir()
+            .context("Failed to read GraalVM cache directory")?
+            .collect_vec();
+        let [graalvm_dir] = dir_entries.as_slice() else {
+            bail!("GraalVM cache directory should contain exactly one directory");
+        };
+        let graalvm_dir = match graalvm_dir {
```

Comment:
Something like
```
let graalvm_dir = graalvm_dir.context("Failed to read GraalVM cache directory.")?;
```

Would be a little bit more idiomatic (context adds additional error "layer", we don't need then to manually merge it with previous error.

Reply:
For now, I decide to ignore this suggestion, as after applying, the borrow checker fails with "cannot move out of `*graalvm_dir` which is behind a shared reference". And I don't really want to bother with this right now. Thanks for the explanation anyway.